ClientApp/src/garden/garden.js

```diff
@@ -0,0 +1,43 @@
+import React, { useState, useEffect } from 'react';
+import { Canvas } from '@react-three/fiber';
+import { OrbitControls } from '@react-three/drei';
+import { Block } from './Block';
+import { gridSize } from './Plants';
+
+const generateGridCoords = (size) => {
+    let arr = [];
+    const helper = Math.floor(size / 2);
+    for (let i = -1 * helper; i < helper + 1; i++) {
+        for (let j = -1 * helper; j < helper + 1; j++) {
+            arr.push([i, -1, j]);
+        }
+    }
+    return arr;
+};
+let groundCoords = generateGridCoords(gridSize);
+
+export const GardenComp = ({ username, onPlaceChosen, onUserNotFound }) => {
+    const [listings, setListings] = useState(null);
+
+    useEffect(() => {
+        const fetchData = async () => {
+            const response = await fetch(`user/listings/${username}`);
+            if (response.ok) {
+                setListings(await response.json());
+            } else {
+                onUserNotFound && onUserNotFound();
+            }
+        };
+        fetchData();
+    }, [username, onUserNotFound]);
+
+    return (<Canvas style={{ height: "400px" }} camera={{position: [2, 3, 7]}}>
+        <OrbitControls />
```

Comment:
A locked camera with fixed zoom max and zoom, fixed camera movement (like 10-20 degrees to each side with locked position) could be added so the user couldn't zoom out of the rendered object.

Reply:
Good idea, but will probably implement next week.